### ISSUE TITLE
Simplification of the <semantics> element

### ISSUE THREAD

**Issue by fred-wang** [mathml-refresh/mathml#100]
The typical usage of the semantics element is as followed:

    <semantics>
      [ANNOTATED PRESENTATION MATHML CHILD]
      <annotation encoding="application/x-tex">...</annotation>
      <annotation-xml encoding="MathML-Content">...</annotation-xml>
      ... MORE ANNOTATIONS ...
    </semantics>

WebKit and Gecko will render the first child.

Currently, Chromium (without MathML support) will always hide that child but display all the annotations (which is ok when there is only one text annotation):
https://chromium.googlesource.com/chromium/src/+/HEAD/third_party/blink/renderer/core/css/mathml.css#40

A non-common case is when the first child is actually content MathML:
https://mathml-refresh.github.io/mathml/chapter5.html#mixing.alternate.representations
WebKit and Gecko will then actually try to render either a text annotation or a MathML/SVG/HTML annoation-xml:
https://mathml-refresh.github.io/mathml-core/#sematicsandpresentation
(Chromium will wrongly display all the children)

Another (non-valid) use case is when the semantics only has one annotation-xml child: 

      <semantics>
        <annotation-xml encloding="SVG1.1">...</annotation-xml>
      </semantics>

which @distler has used to embed SVG inside MathML.
WebKit, Gecko and Chromium will properly handle that case too.

**Comment by fred-wang** [comment]
I'm not sure what will be the outcome of #47, but browser developers are not really fond of content MathML. My suggestion for MathML Core is thus to assume that the first child is presentation markup and then hide the annotations. Hence the proposal will be the same as what we decided for maction (#26):

- Treat maction as an mrow for layout purpose.
- Make the default display values of all but the first child display: none (it can be overridden by users via standard CSS). 

This is Gecko's legacy implementation which is compatible with (and motivated) the non-valid SVG case above. Users could also override the behavior if they want to render annotations instead of presentation MathML.

**Comment by NSoiffer** [comment]
That seems reasonable to me. Core is all about making rendering in browsers
simpler.

A polyfill can always rearrange the children of semantics to make this
work... assuming that there is some presentation that can be moved to be
the first child. Potentially, the polyfill could hook into a
content-to-presentation translator to generate presentation MathML. A
wrinkle on such a translator is that needs to generate core MathML.

**Comment by fred-wang** [comment]
We agreed during yesterday's meeting to follow https://github.com/mathml-refresh/mathml/issues/100#issuecomment-500293044
I've rewritten the section accordingly: https://mathml-refresh.github.io/mathml-core/#semantics-and-presentation

**Comment by fred-wang** [comment]
https://lists.w3.org/Archives/Public/public-mathml4/2019Jun/0013.html

ADOPTED: render 1st child for Core default

ACTION ITEM: ADD issue for the Full spec as to what to do; this touches on
#101 on integration; and how the polyfill is affected

**Comment by Jamesernator** [comment]
> but browser developers are not really fond of content MathML

They should be using it for screen readers, but otherwise I don't see what browsers should do with content MathML.

Content MathML is explicitly defined as not being tied to presentation and may not have a particular rendering ([it's even the first sentence of the spec](https://w3c.github.io/mathml/mathml.html#chapter4_id.4.1.1)) so I don't think it makes sense for browsers attempt to render it.

I personally think it's better to leave it to userland to generate or author a specific rendering as it's more likely the author will have a clearer preference for their specific use case on how it should be rendered.

**Comment by fred-wang** [comment]
Tested by
https://w3c-test.org/mathml/presentation-markup/mrow/legacy-mrow-like-elements-001.html
https://w3c-test.org/mathml/presentation-markup/mrow/legacy-mrow-like-elements-002.html

so closing.

**Comment by NSoiffer** [comment]
Needs a polyfill, so reopening.

**Comment by davidcarlisle** [comment]
> Needs a polyfill, so reopening.

@NSoiffer  I think it probably just needs CSS to display a presentation mathml annotation if it exists, something like this.
We could still use a javascript polyfill to inject the css

```xml
<!DOCTYPE html>
<html>
<head>
  <meta encoding="UTF-8"/>
  <style>
   semantics > annotation-xml[encoding='MathML-Presentation'] {display: block}
   semantics > *:has(+ annotation-xml[encoding='MathML-Presentation']) {display: none}
  </style>
</head>
<body>
<p>presentation first</p>
<math display="block">
 <semantics>
  <mrow><mn>1</mn><mo>+</mo><mi>a</mi></mrow>
  <annotation-xml encoding="MathML-Content">
   <apply><plus/><cn>1</cn><ci>a</ci></apply>
  </annotation-xml>
 </semantics>
</math>

<p>content first</p>
<math display="block">
 <semantics>
  <apply><plus/><cn>1</cn><ci>a</ci></apply>
  <annotation-xml encoding="MathML-Presentation">
   <mrow><mn>1</mn><mo>+</mo><mi>a</mi></mrow>
  </annotation-xml>
 </semantics>
</math>

</body>
</html>
```

**Comment by NSoiffer** [comment]
Yes, this can be done with CSS. I can also be done with JS, or with JS injecting the CSS.

It may make sense to develop a CSS file that people can download. Several of my polyfills require CSS. I'm not sure if we should split a CSS file up into pieces (as the JS polyfills currently are done).

Thoughts?
